tests/cuda/CMakeLists.txt

```diff
@@ -14,6 +14,27 @@ include( vecmem-compiler-options-cuda )
 # External dependency/dependencies.
 find_package( CUDAToolkit REQUIRED )
 
+add_library(
+   vecmem_testing_cuda_main
+   SHARED
+   test_cuda_main.cu
+)
+
+target_link_libraries(
+   vecmem_testing_cuda_main
+   PUBLIC
+   CUDA::cudart
+   gtest
+   vecmem::core
+   vecmem::cuda
```

Comment:
It should not need to link against either of those. The one header file that it uses from `vecmem::cuda` is a private header. So linking against `vecmem::cuda` doesn't even help with that...